gemma.cc

```diff
@@ -813,6 +817,47 @@ hwy::AlignedFreeUniquePtr<uint8_t[]> GetCompressedWeightsT(
   }
 }
 
+template <class TConfig>
+void CompressWeights(const Path& weights_path,
+                     const Path& compressed_weights_path,
+                     hwy::ThreadPool& pool) {
+  if (!std::filesystem::exists(weights_path.path)) {
+    HWY_ABORT("The model weights file '%s' does not exist.",
+              weights_path.path.c_str());
+  }
+
+  // Allocate compressed weights.
+  using CWeights = CompressedWeights<TConfig>;
+  hwy::AlignedFreeUniquePtr<uint8_t[]> c_weights_u8 =
+      hwy::AllocateAligned<uint8_t>(sizeof(CWeights));
+  CWeights* c_weights = reinterpret_cast<CWeights*>(c_weights_u8.get());
+  new (&c_weights->c_layer_ptrs) CompressedLayerPointers<TConfig>(pool);
+
+  // Get weights, compress, and store.
+  const hwy::AlignedUniquePtr<Weights<TConfig>> weights =
+      LoadWeights<TConfig>(weights_path);
+  Compressor compressor(pool);
+  ForEachTensor<TConfig>(weights.get(), *c_weights, compressor);
+  compressor.WriteAll(pool, compressed_weights_path.path.c_str());
+
+  c_weights->c_layer_ptrs.~CompressedLayerPointers<TConfig>();
+}
+
+void CompressWeightsT(gcpp::Model model, const Path& weights,
```

Comment:
Right above this we have a GetCompressedWeightsT, can't we use that?

Reply:
I wanted to add a separate function, because I think the functionality to fall back to uncompressed weights should be removed from GetCompressedWeights, and the --weights flag should be deprecated in gemma and work only in this separate tool. This should not be a problem since I could not find any working uncompressed weights file anywhere.

Reply:
Ah, I see. Yes, agree we can remove the --weights soon. It's possibly useful still during the short term for Phil's export script, which does produce uncompressed weights.